python/lsst/pipe/tasks/insertFakeSources.py

```diff
@@ -0,0 +1,597 @@
+                try:
+                    flux = photoCalib.magnitudeToInstFlux(row[self.config.magVar % band], xy)
+                except LogicError:
+                    flux = 0
+            else:
+                flux = photoCalib.getFlux(row[self.config.magVar % band])
+
+            bulge = galsim.Sersic(row[self.config.nBulge], half_light_radius=row[self.config.bulgeHLR])
+            axisRatioBulge = row[self.config.bBulge]/row[self.config.aBulge]
+            bulge = bulge.shear(q=axisRatioBulge, beta=((90 - row[self.config.paBulge])*galsim.degrees))
+
+            disk = galsim.Sersic(row[self.config.nDisk], half_light_radius=row[self.config.diskHLR])
+            axisRatioDisk = row[self.config.bDisk]/row[self.config.aDisk]
+            disk = disk.shear(q=axisRatioDisk, beta=((90 - row[self.config.paDisk])*galsim.degrees))
+
+            gal = disk + bulge
+            gal = gal.withFlux(flux)
+
+            try:
+                psfKernel = psf.computeKernelImage(xy).getArray()
+            except InvalidParameterError:
+                self.log.info("Galaxy at %0.4f, %0.4f outside of image" % (row["x"], row["y"]))
+                continue
+            psfIm = galsim.InterpolatedImage(galsim.Image(psfKernel), scale=pixelScale)
+            gal = galsim.Convolve([gal, psfIm])
+            try:
+                galIm = gal.drawImage(scale=pixelScale, method="real_space").array
+            except (galsim.errors.GalSimFFTSizeError, MemoryError):
+                continue
+
+            galImages.append((afwImage.ImageF(galIm), xy))
+
+        return galImages
+
+    def mkFakeStars(self, fakeCat, band, photoCalib, psf, calexp):
+
+        """Make fake stars based off the properties in the fakeCat.
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+        band : `str`
+        photoCalib : `lsst.afw.image.calib.Calib` or `lsst.afw.image.photoCalib.PhotoCalib`
+        psf : `lsst.meas.extensions.psfex.psfexPsf.PsfexPsf`
+
+        Returns
+        -------
+        starImages : `list`
+            A list of `lsst.afw.image.image.image.ImageF` of fake stars
+        """
+
+        starImages = []
+
+        self.log.info("Making %d fake star images" % len(fakeCat))
+
+        for (index, row) in fakeCat.iterrows():
+            # jointCal photoCalib and normal photoCalib have different ways of doing this.
+            xy = afwGeom.Point2D(row["x"], row["y"])
+            if self.config.useUpdatedCalibs and self.calexpType == "calexp":
+                # This should go away when this is switched to reference catalogs
+                try:
+                    flux = photoCalib.magnitudeToInstFlux(row[band + "magVar"], xy)
+                except LogicError:
+                    flux = 0
+            else:
+                flux = photoCalib.getFlux(row[band + "magVar"])
+
+            try:
+                starIm = psf.computeImage(xy)
+
+            except InvalidParameterError:
```

Comment:
Same comment above about doing xy check first, although in this case all that would save you is computing the psf I guess.